### PR TITLE
Testfx find place

### PR DESCRIPTION
**Pull request by JonLavi**
Hi Tyler,
As discussed, here’s a draft PR for the testfx experiment.

### CODE REVIEW

File: search/find-place/src/test/java/com/esri/samples/find_place/FindPlaceTest.java [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+    }
+
+    // get a handle on the ComboBoxes
+    placeBox = lookup("#placeBox").queryComboBox();
+    locationBox = lookup("#locationBox").queryComboBox();
+
+    // get a handle on the ComboBoxes' arrow buttons
+    for (Node child : locationBox.getChildrenUnmodifiable()) {
+      if (child.getStyleClass().contains("arrow-button")) {
+        locationBoxArrowRegion = ((Pane) child).getChildren().get(0);
```
Comment (tschie) [inline search/find-place/src/test/java/com/esri/samples/find_place/FindPlaceTest.java:63]: not sure why this is getting set in a loop

File: search/find-place/src/test/java/com/esri/samples/find_place/FindPlaceTest.java [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+public class FindPlaceTest extends FxRobot {
+
+  public static final int SLEEP_500MS = 500;
```
Comment (tschie) [inline search/find-place/src/test/java/com/esri/samples/find_place/FindPlaceTest.java:32]: why are these needed?

File: search/find-place/src/test/java/com/esri/samples/find_place/FindPlaceTest.java [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+  public void panEnablesRedoSearchButton() {
+
+    // get a handle on the Redo Search button
+    Button redoButton = lookup("#redoButton").queryButton();
+
+    // perform a search with pre-defined search inputs
+    clickOn(placeBoxArrowRegion);
+    clickOn("Starbucks");
+
+    clickOn(locationBoxArrowRegion);
```
Comment (tschie) [inline search/find-place/src/test/java/com/esri/samples/find_place/FindPlaceTest.java]: Chain methods that are related. Probably separate out the sleep calls for readability.

File: search/find-place/src/test/java/com/esri/samples/find_place/FindPlaceTest.java [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+    FxToolkit.cleanupStages();
+    // wait before the next test
+    sleep(SLEEP_1000MS);
+  }
+
+  /**
+   * Test Case 1: When using one of the pre-defined search inputs, the app returns the expected number of results
+   */
+  @Test
+  public void defaultSearchInput() {
```
Comment (tschie) [inline search/find-place/src/test/java/com/esri/samples/find_place/FindPlaceTest.java:91]: In automated tests, we never actually want to make real network requests. Instead, we should mock them.

Simply verifying the default options for the dropdowns are present is good enough for this kind of "end-to-end" test. To test that the `search` method on the controller is called, you'd want to use TestFX at the controller level instead of the application level. Then you could use something like Mockito to mock the controller.

File: search/find-place/src/test/java/com/esri/samples/find_place/FindPlaceTest.java [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+
+    // assert that the expected number of graphics (location pins) is displayed
+    assertEquals("Unexpected number of graphics (location pins) found", 50, graphicsOverlay.getGraphics().size());
+  }
+
+  /**
+   * Test Case 2: When using a custom search query, the app returns the expected number of results
+   */
+  @Test
+  public void customSearchInput() {
```
Comment (tschie) [inline search/find-place/src/test/java/com/esri/samples/find_place/FindPlaceTest.java:112]: I have noticed a bug in this sample where if you clear the place input (or edit it significantly), the default options are removed. Would be good to have a test for that and use it to fix the bug and verify.

**Review by tschie — COMMENTED** [review]
Good stuff. The comments I made in your other PR are also mostly relevant here.

File: search/find-place/src/test/java/com/esri/samples/find_place/FindPlaceTest.java (hunk 1 above)
Comment (JonLavi) [reply search/find-place/src/test/java/com/esri/samples/find_place/FindPlaceTest.java:63]: Since there's two ComboBoxes, I needed to find a way to get a handle on each 'arrow-button' and know which is which. Since I think if I simply do `lookup(".arrow-button").queryAll()` I'll get two nodes and not know which ComboBox they belong to? Might be a more elegant way to do this though

File: search/find-place/src/test/java/com/esri/samples/find_place/FindPlaceTest.java (hunk 2 above)
Comment (JonLavi) [reply search/find-place/src/test/java/com/esri/samples/find_place/FindPlaceTest.java:32]: As you pointed out below, we're still doing live server requests here, so I had these sleep intervals put in to wait for the response. They are defined as constants just so they can be changed at one point and it'd apply everywhere

File: search/find-place/src/test/java/com/esri/samples/find_place/FindPlaceTest.java (hunk 4 above)
Comment (JonLavi) [reply search/find-place/src/test/java/com/esri/samples/find_place/FindPlaceTest.java:91]: Yep, I've noticed the tests sometimes tried to proceed before the server response came back. I'll look into mocking!

**Comment by JonLavi** [comment]
closing as there will not likely be any progress on this in the forseable future
